### PR TITLE
Uniwill change to dev branch

### PR DESCRIPTION
**Pull request by rushdie99**
From master to dev.

### CODE REVIEW

File: Installer/installer.iss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#define public Version "0.6.1"
+#define public Version "9.9.9.T3"
```
Comment (diogotr7) [inline Installer/installer.iss]: Please revert this

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+using Aurora.Utils;
+using IronPython.Runtime.Operations;
+using keyboard;
+using Microsoft.Win32;
+using RegistryUtils;
+
+namespace Aurora.Devices.Uni
+{
+    
+    public class KeyboardDevice : Device
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: Please auto format this file. The class should be renamed to be more descriptive. KeyboardDevice gives me no useful information.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        {
+            devicename = KeyboardFactory.GetOEMName();
+          
+            ChoiceGamingCenter();
+ 
+        }
+ 
+     
+
+        private void _CheckControlCenter_Elapsed(object sender, System.Timers.ElapsedEventArgs e)
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: what is this method for?

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+        {
+           
+        }
+
+        private void ChoiceGamingCenter()
+        {
+            GamingCenterType = CheckGC();
+             
+            if (GamingCenterType == 1)
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: GamingCenterType should probably be an enum.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+                    isInitialized = false;
+                    return false;
+                    // Mark Initialized = TRUE
+                   
+                    }
+                    catch (Exception ex)
+                    {
+                        Global.logger.Error("keyboard device, Exception! Message:" + ex);
+                    }
+                    //try
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: does this commented code need to be here ?

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+
+            bRefreshOnce = false;
+            watch.Stop();
+
+            lastUpdateTime = watch.ElapsedMilliseconds;
+
+            return update_result;
+        }
+
+        public bool UpdateDevice(DeviceColorComposition colorComposition, DoWorkEventArgs e, bool forced = false)
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: This method is usually used as a timer proxy for the other UpdateColors method. if you don't need the other component of colorComposirtion, please consult the other device implementations for the usual usage. Repeating the same code in 2 methods doesnt do any good here.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+
+            bool update_result = false;
+
+            watch.Restart();
+
+
+            //Alpha necessary for Global Brightness modifier
+            List<DeviceKeys> KeysList = new List<DeviceKeys>();
+
+            foreach (var key in keyColors.Keys) KeysList.Add(key);
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: why do you loop twice here? 
if you are just changing the brightness of keyColors you should instead do something like
```cs
var adjustedColors = keyColors.Keys.Select(kc => AdjustBrightness(kc));

keyboard.SetEffect(adjustedColors);
```

File: Project-Aurora/Project-Aurora/Project-Aurora.csproj [hunk 8]
```diff
@@ -122,6 +122,10 @@
     <Reference Include="IronPython.Wpf, Version=2.7.8.0, Culture=neutral, PublicKeyToken=7f709c5b713576e1, processorArchitecture=MSIL">
       <HintPath>..\packages\IronPython.2.7.8.1\lib\net45\IronPython.Wpf.dll</HintPath>
     </Reference>
+    <Reference Include="keyboard, Version=1.0.0.0, Culture=neutral, processorArchitecture=AMD64">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>.\keyboard.dll</HintPath>
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Project-Aurora.csproj]: please rename this to something more descriptive.  I will have no idea what "keyboard.dll" does in a few days.

File: Project-Aurora/Project-Aurora/Settings/Configuration.cs [hunk 9]
```diff
@@ -8,7 +8,7 @@
 using Aurora.Profiles.Generic_Application;
 using Aurora.Profiles;
 using Newtonsoft.Json.Serialization;
-
+using keyboard;
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Settings/Configuration.cs]: what is this using for?

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs [hunk 10]
```diff
@@ -0,0 +1,346 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Aurora.Devices.RGBNet;
+using Aurora.Settings;
+using Aurora.Utils;
+using IronPython.Runtime.Operations;
+using keyboard;
+using Microsoft.Win32;
+using RegistryUtils;
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs]: why do you need a different library for this? you could use `Microsoft.Win32.Registry`

Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs:17]: Thanks for your suggestion. I will change it.

File: Project-Aurora/Project-Aurora/Settings/Configuration.cs (hunk 9 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Settings/Configuration.cs:11]: Thanks for reminding me. I am going to remove it.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs (hunk 7 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs:273]: Thanks for your suggestion. I will change it.

File: Installer/installer.iss (hunk 1 above)
Comment (rushdie99) [reply Installer/installer.iss:1]: I will revert it.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs (hunk 2 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs:22]: Got it, thanks. I will change to our name.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs (hunk 3 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs:57]: We will remove it.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs (hunk 5 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs:182]: We don't need to use Global.logger. I will remove it.

File: Project-Aurora/Project-Aurora/Project-Aurora.csproj (hunk 8 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Project-Aurora.csproj:127]: I will change to our name.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs (hunk 6 above)
Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs:291]: I will  combine to one method.

File: Project-Aurora/Project-Aurora/Project-Aurora.csproj (hunk 8 above)
Comment (diogotr7) [reply Project-Aurora/Project-Aurora/Project-Aurora.csproj:127]: please remove the reference to the older dll

File: Project-Aurora/Project-Aurora/Devices/Uniwill/UniwillDevice.cs [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+
+        private System.Diagnostics.Stopwatch watch = new System.Diagnostics.Stopwatch();
+        private long lastUpdateTime = 0;
+
+        System.Timers.Timer regTimer;
+        const string Root = "HKEY_LOCAL_MACHINE";
+        const string subkey = @"SOFTWARE\OEM\Aurora";
+        const string keyName = Root + "\\" + subkey;
+        int SwitchOn = 0;
+        List<AuroraInterface> DeviceList = new List<AuroraInterface>();
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/UniwillDevice.cs:42]: this isn't being used, what is it for?

Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/UniwillDevice.cs:42]: Actually, We have other devices, but it’s just not ready right now.

File: Project-Aurora/Project-Aurora/Devices/Uniwill/UniwillDevice.cs [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+     
+        private AuroraInterface keyboard = null;
+     
+        System.Timers.Timer _CheckControlCenter = new System.Timers.Timer();
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Devices/Uniwill/UniwillDevice.cs:46]: this is also not being used

Comment (rushdie99) [reply Project-Aurora/Project-Aurora/Devices/Uniwill/UniwillDevice.cs:46]: Yes, I’ll remove both. Thanks you.

**Comment by diogotr7** [comment]
I have resolved the merge conflict and made some whitespace and code style changes locally, which will not affect any of the functionality. Is it okay if i push these changes here to finally merge into dev? @rushdie99

**Pull request merged**